### DISCUSSION TITLE
Support for custom External Login Providers?

### DISCUSSION THREAD

**Discussion by lanthonyneville** [oqtane/oqtane.framework#4220] [Q&A] [answered]
Are there any plans to support custom (i.e. installable) external login providers?

Currently there are 2 external login providers: OpenID Connect and OAuth 2.0, which cover a lot of scenarios, but some of us need to be able to authenticate against other sources, e.g. the company Active Directory, or an obscure hand rolled identity system coded into a legacy application years ago (cough ... I'm just asking for a friend OK?).

I notice that the external login provider system currently has no support for additional "plugin" providers - the above 2 are hard coded in.

Is there any plan to support this in the future? If not, I guess the way to implement a custom provider is to make a login module that checks the credentials against the source you need, then creates & logs in the user in the same way that the standard Oqtane login module does. Any other suggestions?

References: [Oqtane 3.1 adds external login support via OAuth2 / OpenID Connect](https://github.com/oqtane/oqtane.framework/discussions/2061), and [Blog post on OAuth 2.0 and OpenId Connect](https://www.oqtane.org/blog/!/37/oauth-2-0-and-openid-connect)

**Comment by sbwalker** [comment, marked as answer]
@lanthonyneville there is no plan for Oqtane to support custom Login/Authentication providers
